File: Online Judges/CodeForces/Round E58 D2/e.py
min_x = 0
min_y = 0
ans = ""
for i in range(int(input())):
    q, _x, _y = input().split()
    _x = int(_x)
    _y = int(_y)
    x = min(_x, _y)
    y = max(_x, _y)
    if q == "+":
        min_x = max(x, min_x)
        min_y = max(y, min_y)
    else:
        if (x >= min_x and y >= min_y):
            ans += "YES\n"
        else:
            ans += "NO\n"
print(ans, end="") 

File: Online Judges/CodeForces/Round E58 D2/e.cpp
#include <iostream>
#include <stdio.h>

using namespace std;

int main() {

    int n;
    scanf("%d\n", &n);

    int max_width = 0, max_height = 0;
    for (int i = 0; i < n; ++i) {
        int x, y, n, m;
        char c;
        scanf("%c %d %d\n", &c, &n, &m);
        x = min(n, m);
        y = max(n, m);
        if (c == '+') {
            max_width = max(x, max_width);
            max_height = max(y, max_height);
        } else {
            if (x >= max_width && y >= max_height) {
                printf("YES\n");
            } else {
                printf("NO\n");
            }
        }
    }

    return 0;
}